src/pages/WalletsMigration/WalletsMigration.js

```diff
@@ -0,0 +1,127 @@
+// @flow strict
+
+import Promise from 'bluebird'
+import React, { Component } from 'react'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { withI18n } from '@lingui/react'
+import { type I18n } from '@lingui/core'
+
+import {
+  Form,
+  type FormRenderProps,
+} from 'react-final-form'
+
+import { migrate } from 'store/migrations'
+import { gaSendEvent } from 'utils/analytics'
+import { WalletPasswordForm } from 'components'
+
+import styles from './walletsMigration.m.scss'
+
+type Props = {|
+  +i18n: I18n,
+  +hint: string,
+|}
+
+const INITIAL_VALUES: FormFields = {
+  password: '',
+}
+
+class WalletsMigration extends Component<Props> {
+  handleSubmit = async (values: FormFields): Promise<?FormFields> => {
+    const { i18n }: Props = this.props
+
+    gaSendEvent(
+      'MigrateWallets',
+      'PasswordEntered',
+    )
+
+    try {
+      await migrate(values.password || '')
+
+      gaSendEvent(
+        'MigrateWallets',
+        'WalletsMigrationSuccess',
+      )
+
+      window.location.reload()
+
+      return null
+    } catch (error) {
+      console.error(error)
+
+      gaSendEvent(
+        'MigrateWallets',
+        'WalletsMigrationError',
+      )
+
+      return {
+        password: i18n._(
+          'entity.Password.error.invalid',
+          null,
+          { defaults: 'Invalid password' },
+        ),
+      }
+    }
+  }
+
+  renderForm = ({
+    handleSubmit,
+    values = {},
+    submitting,
+  }: FormRenderProps) => {
+    const { i18n }: Props = this.props
+
+    return (
+      <WalletPasswordForm
+        handleSubmit={handleSubmit}
+        values={values}
+        hint={this.props.hint}
+        description={i18n._(
+          'WalletsMigration.description',
+          null,
+          // eslint-disable-next-line max-len
+          { defaults: 'To migrate storage data, you need to enter security password. Please don\'t close the tab, this process will take a while.' },
+        )}
+        isSubmitting={submitting}
+      />
+    )
+  }
+
+  render() {
+    return (
+      <div className={styles.core}>
+        <Form
+          render={this.renderForm}
+          onSubmit={this.handleSubmit}
+          initialValues={INITIAL_VALUES}
+        />
+      </div>
+    )
+  }
+}
+
+function mapStateToProps(state: AppState) {
+  const passwordVersion: number = state.password && state.password.persist
+    // $FlowFixMe
+    ? state.password.persist.version
+    : 0
+
+  return {
+    hint: passwordVersion
+      ? state.password.persist.hint
+      // $FlowFixMe
+      : (state.wallets && state.wallets.persist && state.wallets.persist.passwordOptions)
+        ? state.wallets.persist.passwordOptions.passwordHint
+        : '',
+  }
```

Comment:
I suggest we avoid nested ternaries whenever we can.

This one could be rewritten as:
```
if (passwordVersion) {
  return state.password.persist
}

return {
  hint: (state.wallets && state.wallets.persist && state.wallets.persist.passwordOptions)
    ? state.wallets.persist.passwordOptions.passwordHint
    : '',
}
```